sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapp/AADOAuth2UserService.java

```diff
@@ -62,6 +68,13 @@ public OidcUser loadUser(OidcUserRequest userRequest) throws OAuth2Authenticatio
             return (DefaultOidcUser) session.getAttribute(DEFAULT_OIDC_USER);
         }
 
+        if (idToken.containsClaim(ROLES)) {
+            roles = idToken.getClaimAsStringList(ROLES)
```

Comment:
Please also check for NULL. According to [docs](https://docs.spring.io/spring-security/site/docs/current/api/org/springframework/security/oauth2/core/ClaimAccessor.html#getClaimAsStringList-java.lang.String-) this method could return NULL if the claim doesn't exist or is not a list.

Reply:
@yiliuTo How about write like this:
```
        Optional.ofNullable(idToken)
                .map(token -> token.getClaimAsStringList(ROLES))
                .map(Collection::stream)
                .orElseGet(Stream::empty)
                .filter(StringUtils::hasText)
                .map(role -> APPROLE_PREFIX + role)
                .forEach(roles::add);
```

And please add test about:
 1. claim does not exist.
 2. cannot be assigned to a list